### PR TITLE
Change all calls that use fmt.Errorf and an err, to errors.Wrapf

### PR DESCRIPTION
**Pull request by rhatdan**
We want to wrap all err calls with Wrapf, to allow for easier debugging
and better error reporting.

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes: https://github.com/projectatomic/buildah/issues/89

File: run.go [hunk 1]
```diff
@@ -151,7 +151,7 @@ func (b *Builder) Run(command []string, options RunOptions) error {
 		spec.Process.Cwd = DefaultWorkingDir
 	}
 	if err = os.MkdirAll(filepath.Join(mountPoint, b.WorkDir()), 0755); err != nil {
-		return fmt.Errorf("error ensuring working directory %q exists: %v)", b.WorkDir(), err)
+		return errors.Wrapf(err, "error ensuring working directory %q exists", b.WorkDir(), err)
```
Comment (TomSweeneyRedHat) [inline run.go]: I think you want to remove the ',err' at the end of this line.

Comment (rhatdan) [inline run.go]: Yup, surprised compiler did not catch that thanks.

**Comment by nalind** [comment]
Can you add fixes for `util/util.go`, `imagebuildah/build.go`, and `imagebuildah/util.go`?

File: cmd/buildah/mount.go [hunk 2]
```diff
@@ -41,21 +42,20 @@ func mountCmd(c *cli.Context) error {
 	}
 
 	if len(args) == 1 {
-		name = args[0]
-
+		name := args[0]
```
Comment (nalind) [inline cmd/buildah/mount.go]: Need to either revert this line, or remove the declaration of `name` at the start of `main()`.

Comment (rhatdan) [inline cmd/buildah/mount.go]: Fixed

**Comment by rhatdan** [comment]
I think I got all of the code now.

**Comment by nalind** [comment]
The commit log message appears to describe #124.

**Comment by rhatdan** [comment]
Fixed commit message.

File: user_basic.go [hunk 3]
```diff
@@ -7,13 +7,13 @@ import (
 )
 
 func lookupUserInContainer(rootdir, username string) (uint64, uint64, error) {
-	return 0, 0, fmt.Errorf("user lookup not supported")
+	return 0, 0, errors.Wrapf("user lookup not supported")
```
Comment (nalind) [inline user_basic.go]: Should this be passing that error message to `errors.New()` instead?

File: user_unix_cgo.go [hunk 4]
```diff
@@ -31,13 +33,13 @@ func fopenContainerFile(rootdir, filename string) (C.pFILE, error) {
 	defer C.free(unsafe.Pointer(mode))
 	f, err := C.fopen(cctrfile, mode)
 	if f == nil || err != nil {
-		return nil, fmt.Errorf("error opening %q: %v", ctrfile, err)
+		return nil, errors.Wrapf(err, "error opening %q", ctrfile)
 	}
 	if err = syscall.Fstat(int(C.fileno(f)), &st); err != nil {
-		return nil, fmt.Errorf("fstat(%q): %v", ctrfile, err)
+		return nil, errors.Wrapf(err, "fstat(%q): %v", ctrfile)
```
Comment (nalind) [inline user_unix_cgo.go]: Need to drop the `: %v` from this error message.

Comment (rhatdan) [inline user_unix_cgo.go]: Fixed

**Comment by nalind** [comment]
LGTM
@rh-atomic-bot r+ 3db35b69016585e7c82cf95b2da9963b13adda10

**Pull request closed without merging**